src/python/espressomd/visualization.pyx

```diff
@@ -0,0 +1,209 @@
+        self.last_N = 1
+        self.last_Nbonds = 1
+        self.last_boxl = [0,0,0]
+        self.running = False
+        self.last_T = -1
+
+        # GUI window
+        self.gui = GUI()
+        self.timers = [Timer(100, self._draw)]
+
+    def _draw(self):
+        """Update the Mayavi objects with new particle information.
+        This is called periodically in the GUI thread"""
+        if self.data is None:
+            return
+        
+        assert isinstance(threading.current_thread(), threading._MainThread)
+
+        f = mlab.gcf()
+        visual.set_viewer(f)
+
+        coords, types, radii, N_changed, bonds, Nbonds_changed, boxl, box_changed = self.data
+        self.data = None
+
+        if box_changed or not self.running:
+            self.box.set(bounds=(0,boxl[0], 0,boxl[1], 0,boxl[2]))
+
+        if not N_changed:
+            self.points.mlab_source.set(x=coords[:,0]%boxl[0], y=coords[:,1]%boxl[1], z=coords[:,2]%boxl[2], u=radii, v=radii, w=radii, scalars=types)
+        else:
+            self.points.mlab_source.reset(x=coords[:,0]%boxl[0], y=coords[:,1]%boxl[1], z=coords[:,2]%boxl[2], u=radii, v=radii, w=radii, scalars=types)
+        if not self.running:
+            f.scene.reset_zoom()
+            self.running = True
+
+        if not Nbonds_changed:
+            if bonds.shape[0] > 0:
+                self.arrows.mlab_source.set  (x=bonds[:,0], y=bonds[:,1], z=bonds[:,2], u=bonds[:,3], v=bonds[:,4], w=bonds[:,5])
+        else:
+            self.arrows.mlab_source.reset(x=bonds[:,0], y=bonds[:,1], z=bonds[:,2], u=bonds[:,3], v=bonds[:,4], w=bonds[:,5], scalars=bonds[:,6])
+
+    def update(self):
+        """Pull the latest particle information from Espresso.
+        This is the only function that should be called from the computation thread.
+        It does not call any Mayavi functions unless it is being called from the main (GUI) thread."""
+
+        if self.last_T is not None and self.last_T == self.system.time:
+            return
+        self.last_T = self.system.time
+        
+        cdef int N = self.system.n_part
+        coords = numpy.zeros((N,3))
+        types = numpy.empty(N, dtype=int)
+        inter = NonBondedInteractions()
+        radii = numpy.empty(N)
+        cdef int i=0,j=0,k=0 
+        cdef int t 
+        cdef int partner
+        cdef particle p
+        cdef ia_parameters* ia
+        cdef vector[int] bonds
+
+    # Using (additional) untyped variables and python constructs in the loop 
+    # will slow it down considerably. 
+        for i in range(self.system.max_part+1):
+            if not particle_exists(i):
+                continue
+            get_particle_data(i,&p)
+            coords[j,:] = p.r.p
+            t = p.p.type
+            types[j] = t +1
+            radii[j] =get_ia_param(t,t).LJ_sig *0.5
+
+            # ITerate over bonds
+            k=0        
+            while k<p.bl.n:
+                # Bond type
+                t= p.bl.e[k]
```

Comment:
You should store `t` into the bonds array so that we can color the bonds based on bond type.